compiler-rt/lib/builtins/extendhfxf2.c

```diff
@@ -0,0 +1,18 @@
+//===-- lib/extendhfxf2.c - half -> long double conversion --------*- C -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#define SRC_HALF
+#define DST_DOUBLE
```

Comment:
I guess this works fine since you are doing an explicit double->long double extension and I think all hf values should be representable.

I do wonder if it would be cleaner to update fp_extend.h to handle DST_80 instead and do the following:
```suggestion
#define DST_80
```

Reply:
Because of the integer part at bit 63, it would require some changes in extendxfyf2.  But with type casting `flds` instruction takes care of the 80bit extended format.

Reply:
Ah if it's nontrivial, then the explicit cast seems fine.